Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Post Item/Components/Post Interaction Bar.swift

```diff
@@ -0,0 +1,149 @@
+//
+
+import SwiftUI
+
+import Foundation
+
+/**
+ View grouping post interactions--upvote, downvote, save, reply, plus post info
+ */
+struct PostInteractionBar: View {
+    // constants
+    let iconToTextSpacing: CGFloat = 2
+    let iconPadding: CGFloat = 4
+    let iconCorner: CGFloat = 2
+    let scoreItemWidth: CGFloat = 12
+    
+    // state fakers--these let the upvote/downvote/score/save views update instantly even if the call to the server takes longer
+    @State var dirtyVote: ScoringOperation
+    @State var dirtyScore: Int
+    @State var dirtySaved: Bool
+    @State var dirty: Bool
+    
+    // computed properties--if dirty, show dirty value, otherwise show post value
+    var displayedVote: ScoringOperation { dirty ? dirtyVote : post.myVote ?? .resetVote }
+    var displayedScore: Int { dirty ? dirtyScore : post.counts.score }
+    
+    // parameters
+    let post: APIPostView
+    let account: SavedAccount
+    let compact: Bool
+    let voteOnPost: (ScoringOperation) async -> Bool
+    
+    init(post: APIPostView, account: SavedAccount, compact: Bool, voteOnPost: @escaping (ScoringOperation) async -> Bool) {
+        self.post = post
+        self.account = account
+        self.compact = compact
+        self.voteOnPost = voteOnPost
+        _dirtyVote = State(initialValue: post.myVote ?? .resetVote)
+        _dirtyScore = State(initialValue: post.counts.score)
+        _dirtySaved = State(initialValue: false)
+        _dirty = State(initialValue: false)
+     }
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            HStack(spacing: 8) {
+                VoteComplex(vote: displayedVote, score: displayedScore, upvote: upvote, downvote: downvote)
+                    .padding(.trailing, 8)
+                // TODO: change all this once saving is implemented
+                SaveButton(saved: false)
+                    .onTapGesture {
+                        Task(priority: .userInitiated) {
+                            await savePost()
+                        }
+                    }
+                ReplyButton()
+                Spacer()
+                infoBlock
+            }
+            .padding(.horizontal)
+            .padding(.vertical, compact ? 2 : 6)
+        }
+        .dynamicTypeSize(compact ? .small : .medium)
+    }
+    
+    // subviews
+
+    var infoBlock: some View {
+        // post info component
+        HStack(spacing: 8) {
+            HStack(spacing: iconToTextSpacing) {
+                Image(systemName: "clock")
+                Text(getTimeIntervalFromNow(date: post.post.published))
+            }
+            HStack(spacing: iconToTextSpacing) {
+                Image(systemName: "bubble.left")
+                Text(String(post.counts.comments))
+            }
+        }
+        .foregroundColor(.secondary)
+    }
+    
+    // helper functions
+    
+    func upvote() async -> Void {
+        fakeUpvote()
+        await voteOnPost(.upvote)
```

Comment:
this returns a `Bool` but we're not using it, does the function need to return a value?

if we don't need a value back, should the `voteOnPost` method instead be declared as:
```swift
func voteOnPost(inputOp: ScoringOperation) async {
    let operation = post.myVote == inputOp ? ScoringOperation.resetVote : inputOp
    try? await ratePost(postId: post.id, operation: operation, account: account, postTracker: postTracker, appState: appState)
}
```